### ISSUE TITLE
Poetry very slow with some packages since 1.2

### ISSUE THREAD

**Issue by jc-louis** [python-poetry/poetry#6906]
Hello,

Since updating to Poetry 1.2.X dependency resolution is very slow for some packages. (After uninstalling 1.2.2 and installing 1.1.15 everything is working fine).

I've tried clearing caches. It seems to always block at `Creating new session for files.pythonhosted.org`.

Exemple with `numpy`: (using `poetry run pip install numpy` works fine)

```
$ poetry init
$ poetry add numpy -vvv

Loading configuration file /Users/jc/Library/Preferences/pypoetry/config.toml
Using virtualenv: /Users/jc/projects/lifen/git/tmp/test-poetry1.2/.venv
Project environment contains an empty path in sys_path, ignoring.
[keyring.backend] Loading KWallet
[keyring.backend] Loading SecretService
[keyring.backend] Loading Windows
[keyring.backend] Loading chainer
[keyring.backend] Loading libsecret
[keyring.backend] Loading macOS
Creating new session for pypi.org
[urllib3.connectionpool] Starting new HTTPS connection (1): pypi.org:443
[urllib3.connectionpool] https://pypi.org:443 "GET /simple/numpy/ HTTP/1.1" 304 0
Source (PyPI): 90 packages found for numpy *
Using version ^1.23.4 for numpy

Updating dependencies
Resolving dependencies...
   1: fact: test-poetry1-2 is 0.1.0
   1: derived: test-poetry1-2
   1: fact: test-poetry1-2 depends on numpy (^1.23.4)
   1: selecting test-poetry1-2 (0.1.0)
   1: derived: numpy (>=1.23.4,<2.0.0)
Source (PyPI): 1 packages found for numpy >=1.23.4,<2.0.0
Source (PyPI): The cache for numpy 1.23.4 is outdated. Refreshing.
Source (PyPI): Getting info for numpy (1.23.4) from PyPI
Source (PyPI): No dependencies found, downloading archives
Source (PyPI): Downloading sdist: numpy-1.23.4.tar.gz
Creating new session for files.pythonhosted.org

→ hangs for a very long time
```

**Comment by mayteio** [comment]
@jc-louis what was your solution/unblocker here? We have the same issue. It hangs on creating a connection pool with pipy:
```
❯❯❯ poetry add pytest@6.2.5 -vvv                                                                                 PACH-5463 🚩
Loading configuration file /Users/harley/Library/Preferences/pypoetry/config.toml
Loading configuration file /Users/harley/Library/Preferences/pypoetry/auth.toml
Adding repository pachama (https://infra-pypicloud.prod.pachama.com/simple)
Using virtualenv: /Users/harley/pachama/marketplace-supply/.venv
Project environment contains an empty path in sys_path, ignoring.
[keyring.backend] Loading KWallet
[keyring.backend] Loading SecretService
[keyring.backend] Loading Windows
[keyring.backend] Loading chainer
[keyring.backend] Loading libsecret
[keyring.backend] Loading macOS
Creating new session for infra-pypicloud.prod.pachama.com
[urllib3.connectionpool] Starting new HTTPS connection (1): infra-pypicloud.prod.pachama.com:443
[urllib3.connectionpool] https://infra-pypicloud.prod.pachama.com:443 "GET /simple/pytest/ HTTP/1.1" 302 199
[urllib3.connectionpool] Starting new HTTPS connection (1): pypi.org:443
[urllib3.connectionpool] https://pypi.org:443 "GET /simple/pytest/ HTTP/1.1" 200 25583
```

**Comment by jc-louis** [comment]
@mayteio contrary to what I said in my initial version it does not hangs forever but it's just extremely slow, I still have the same issue and I just wait...

**Comment by agossino** [comment]
Hello,
I have the same problem with Poetry (version 1.2.2). For me hangs forever but I did not wait more then one hour.


$ poetry add -vvv --group dev pytest
Loading configuration file /home/ago/.config/pypoetry/config.toml
Using virtualenv: /home/ago/.cache/pypoetry/virtualenvs/sb-extractor-3V-FgV4H-py3.9
Project environment contains an empty path in sys_path, ignoring.
[keyring.backend] Loading KWallet
[keyring.backend] Loading SecretService
[keyring.backend] Loading Windows
[keyring.backend] Loading chainer
[keyring.backend] Loading libsecret
[keyring.backend] Loading macOS
Creating new session for pypi.org
[urllib3.connectionpool] Starting new HTTPS connection (1): pypi.org:443
[urllib3.connectionpool] https://pypi.org:443 "GET /simple/pytest/ HTTP/1.1" 200 25711

On the same computer but different account, it works fine and fast. So I think it is a configuration problem. I do not remember what I have changed ...
The same happend if I modify pyproject.toml and then run poetry install. It hangs on pypi connection.

**Comment by dimbleby** [comment]
mostly this symptom is because of a stray `.lock` file in your cache and will be resolved by `find ~/.cache/pypoetry/ -name '*.lock' -delete` or equivalent if your cache is somewhere else.

this is fixed in 1.3

**Comment by agossino** [comment]
@dimbleby solved!! More then 200 .lock files found and deleted. Thank you

**Comment by dimbleby** [comment]
@jc-louis you've reopened this but as it stands it is completely unactionable.

Are you sure this isn't the same problem with .lock files, as above?

If not, and if you're able to reproduce this reliably, then a docker container holding that reproduction so that others can investigate it would be valuable.

But if all we have is "it's slow for me, I can't tell you anything about why" then there's nothing anyone can do with this but close it.

**Comment by ddejohn** [comment]
I just installed poetry 1.2 via pipx in a pyenv-managed Python 3.12.0a1 environment in a brand new WSL2 Ubuntu install. Tried `poetry install pandas` and it hanged for about 12 minutes before exiting with an error about failing to build the wheel.

Despite it being a completely brand new environment, I tried clearing the cache as well as any lock files as suggested above. Not sure what I can do to help / what logs I could dump but this is a pretty disappointing out-of-the-box experience.

**Comment by neersighted** [comment]
There are no wheels available for Python 3.12, and you are building large packages with native code from source. This is a tradeoff you accept when you use an unreleased Python version, and is not related to Poetry.

**Comment by AIGeneratedUsername** [comment]
I can confirm that Poetry 1.2.2 hangs forever for `poetry lock` on Python 3.10.6. 

I was waiting for a few hours. Then I enabled verbosity and I saw that Poetry hangs right after `Creating new session for``` (GitLab in my case). Poetry gets 200 response status code and hangs forever. Nothing happens.

The solution is to run `sudo rm -r ~/.cache/pypoetry`.

**Comment by jc-louis** [comment]
fixed in 1.3 for me, thanks
